pkg/client/apiutil/apimachinery.go

```diff
@@ -169,3 +170,20 @@ func createRestConfig(gvk schema.GroupVersionKind, isUnstructured bool, baseConf
 
 	return cfg
 }
+
+// BoolPointerFlagFunc returns a function to be associated with the custom bool flag func
+//
+// Example:
+//
+// var fooPtr *bool
+// flag.Func("foo", "foo usage", apiutil.BoolPointerFlagFunc(fooPtr))
+func BoolPointerFlagFunc(ptr *bool) func(flagVal string) error {
+	return func(flagVal string) error {
```

Comment:
Shouldn't we say `if flagVal == "" { return nil }` ?

Reply:
I think instead of returning nil, we should assign nil to the bool pointer. Which would imply that
- If the user gave the flag but with empty value, it will be considered as flag not passed
- If the user gave the flag but with some invalid value, it will be error
-  If the user gave the flag but with some valid value, it will be handled properly